src/couch_replicator_connection.erl

```diff
@@ -171,6 +175,33 @@ delete_worker(Worker) ->
     ok.
 
 
+-spec incr_monitor(pid(), #state{}) -> #state{}.
+incr_monitor(Owner, State) ->
+    NewMonitors = case dict:find(Owner, State#state.monitors) of
+        error ->
+            MRef = monitor(process, Owner),
+            dict:store(Owner, {1, MRef}, State#state.monitors);
```

Comment:
Oh, I see what you're saying. That is conceptually simpler. I'll make that change.